pkg/config/remote/service/client_predicates.go

```diff
@@ -0,0 +1,135 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/Masterminds/semver"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+type tracerPredicates struct {
+	RuntimeID     *string `json:"runtime-id,omitempty"`
+	Service       *string `json:"service,omitempty"`
+	Environment   *string `json:"environment,omitempty"`
+	AppVersion    *string `json:"app-version,omitempty"`
+	TracerVersion *string `json:"tracer-version,omitempty"`
+	Language      *string `json:"language,omitempty"`
+}
+
+type clientPredicates struct {
+	Version    int                 `json:"version,omitempty"`
+	Predicates []*tracerPredicates `json:"predicates,omitempty"`
+}
+
+type DirectorTargetsCustomMetadata struct {
+	Predicates *clientPredicates `json:"tracer-predicates,omitempty"`
+}
+
+// Given the hostname and state will parse predicates and execute them
+// It will return a list ConfigPointers
```

Comment:
```suggestion
// It will return a list of paths of the files whose predicates matched
// the client details.
```

We could also just call it ClientConfig which is what the field's name is now, but even `ClientConfig` doesn't really give much detail on what it represents.